service/handlers.go

```diff
@@ -0,0 +1,149 @@
+package service
+
+import (
+	"encoding/json"
+	"fmt"
+	log "github.com/Sirupsen/logrus"
+	"github.com/rancher/go-rancher/v2"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+type WebhookRequest map[string]interface{}
+
+func Construct(w http.ResponseWriter, r *http.Request) {
+	var webhookRequestData WebhookRequest
+	log.Infof("Construct Payload")
+	bytes, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		log.Errorf("Construct failed with error: %v", err)
+		return
+	}
+	json.Unmarshal(bytes, &webhookRequestData)
+	_, err = ConstructPayload(webhookRequestData)
+}
+
+// ConstructPayload accepts and sends request to driver
+func ConstructPayload(parameters map[string]interface{}) (map[string]interface{}, error) {
+	cattleUrl := "http://localhost:8080/v2-beta"
+	webhookType := parameters["driver_type"]
+	switch webhookType {
+	case "service":
+		serviceId := parameters["service_id"].(string)
+		serviceAccountId := parameters["service_account_id"].(string)
+		url := fmt.Sprintf("%s/projects/%s/services/%s", cattleUrl, serviceAccountId, serviceId)
+		parameters["url"] = url
```

Comment:
The payload shouldnt have the url for accessing the service in it. Just the service and project ID and then the execute method should construct a new client using it.